.github/workflows/create_publish_artifacts.yml

```diff
@@ -0,0 +1,63 @@
+name: Build and Package
+on:
+  schedule:
+    - cron: '0 0 * * *'  # Runs the workflow at midnight every day
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  # build and run for RISCV
+  run_riscv_m1_nightly_package:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+       # installs tools, ninja, installs llvm and sets up sccahe
+      - name: setup ubuntu
```

Comment:
I think consistency between the style of stage names would be nice. Here we have `setup` and `setup` and `build` but then `Download` and `Clone` and `Setup`.